src/resourcemanager/resourcemanager.hpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef RESOURCEMANAGER_HPP_
+#define RESOURCEMANAGER_HPP_
+
+#include <mutex>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <aos/sm/resourcemanager.hpp>
+
+namespace aos::sm::resourcemanager {
+
+/**
+ * JSON provider.
+ */
+class JSONProvider : public sm::resourcemanager::JSONProviderItf {
```

Comment:
JSONProvider should be in aos_core_common_cpp as separate module. It will parse json for other aos types as well.